src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/reducers/__tests__/chartBuilderReducer.test.ts

```diff
@@ -625,14 +614,106 @@ describe('chartBuilderReducer', () => {
     });
   });
 
+  describe('UPDATE_MAP_BOUNDARY_LEVELS', () => {
+    const initialState: ChartBuilderState = {
+      axes: {},
+      options: {
+        height: 300,
+        subtitle: '',
+        title: '',
+        titleType: 'default',
+        alt: '',
+      },
+      map: {
+        dataSetConfigs: [
+          {
+            dataSet: { filters: [] },
+            dataGrouping: { type: 'EqualIntervals', customGroups: [] },
+          },
+        ],
+      },
+    };
+
+    test('sets the boundary level to the payload boundaryLevel', () => {
+      const action: ChartBuilderActions = {
+        type: 'UPDATE_MAP_BOUNDARY_LEVELS',
+        payload: {
+          boundaryLevel: 2,
+          dataSetConfigs: [{ dataSet: { filters: [] }, boundaryLevel: 1 }],
+        },
+      };
+
+      const nextState = produce(
+        chartBuilderReducer({
+          data: [],
+          meta: testFullTable.subjectMeta,
+          tableTitle: undefined,
+        }),
+      )(initialState, action);
+
+      expect(nextState.options?.boundaryLevel).toEqual(2);
+      expect(nextState.map?.dataSetConfigs[0]?.boundaryLevel).toEqual(1);
+    });
+  });
```

Comment:
It'd be a good idea to have another test case that checks data set boundary levels don't overwrite any other other part of the data set (i.e. the data grouping)